src/math.rs

```diff
@@ -67,3 +67,137 @@ impl Cube {
         )
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct OrientedBeam {
+    // Translation and orientation together form the transform from "beam coordinates" into world coordinates
+    pub rotation: Quaternion<f64>,
```

Comment:
could it be coupled, consisting in Decomposed<Vector3<f64>, Quaternion<f64>>?

Reply:
Yes, I think that makes sense.